hppc/src/main/templates/com/carrotsearch/hppc/sorted/SortedIterationKTypeVTypeHashMap.java

```diff
@@ -0,0 +1,497 @@
+/*! #set($TemplateOptions.ignored = ($TemplateOptions.isKTypeAnyOf("DOUBLE", "FLOAT", "BYTE"))) !*/
+package com.carrotsearch.hppc.sorted;
+
+import com.carrotsearch.hppc.*;
+import com.carrotsearch.hppc.comparators.*;
+import com.carrotsearch.hppc.cursors.*;
+import com.carrotsearch.hppc.predicates.*;
+import com.carrotsearch.hppc.procedures.*;
+import com.carrotsearch.hppc.sorting.IndirectSort;
+
+/*! #if ($TemplateOptions.KTypeGeneric) !*/
+import java.util.Comparator;
+/*! #end !*/
+import java.util.Iterator;
+import java.util.function.IntBinaryOperator;
+
+/*! #if ($TemplateOptions.anyGeneric) @SuppressWarnings("unchecked") #end !*/
+/*! ${TemplateOptions.generatedAnnotation} !*/
+public class SortedIterationKTypeVTypeHashMap<KType, VType>
+  implements KTypeVTypeMap<KType, VType>
+{
+  public final KTypeVTypeHashMap<KType, VType> delegate;
+  public final int[] iterationOrder;
+
+  public SortedIterationKTypeVTypeHashMap(KTypeVTypeHashMap<KType, VType> delegate) {
```

Comment:
I miss javadoc in this class. I'll add that soon.
Here this constructor is used to sorted by natural ordering.